MetaMask/metamask-extension
# update balances with a single call on bsc,polygon & kcc (CLA & lint issue fixed)

Fixes:

None

Explanation:

In the [current version of metamask](https://github.com/MetaMask/metamask-extension/blob/v10.9.2/app/scripts/lib/account-tracker.js#L199), if you are using ethereum and its testnets, updating the balances of all created accounts can be done in a single rpc call through the [balance-checker](https://github.com/wbobeirne/eth-balance-checker) contract.

As the balance-checker contract is also available on [BSC @ 0x2352c63A83f9Fd126af8676146721Fa00924d7e4](https://bscscan.com/address/0x2352c63A83f9Fd126af8676146721Fa00924d7e4#code), [Polygon @ 0x2352c63A83f9Fd126af8676146721Fa00924d7e4 ](https://polygonscan.com/address/0x2352c63A83f9Fd126af8676146721Fa00924d7e4#code) and [KCC @ 0x675E1d119Fc20E7b26eE23C4b9a83f657ca8c4B3](https://scan.kcc.io/address/0x675E1d119Fc20E7b26eE23C4b9a83f657ca8c4B3/contracts), this kind of optimization can also be applied to these chains.

Manual testing steps:

    Create multiple accounts in metamask, each account with some random amount of native token on that chain.
    Change the amount of native token in each account a little bit
    See if metamask shows the right balances
